packages/core/src/Element.ts

```
@@ -27,12 +27,13 @@ export interface ElementOptions {
 	testObserverFactory?: (t: TestObserver) => TestObserver
 	persistentRunner: boolean
 	testCommander?: TestCommander
+	failStatusCode?: number
 }
 
-export function runUntilExit(fn: () => Promise<void>) {
+export function runUntilExit(fn: () => Promise<number>) {
 	fn()
-		.then(() => {
-			process.exit(0)
+		.then((code: number) => {
+			process.exit(code)
```

If this is failStatusCode shouldn't this only get called within catch?

---

You mean we throw the `failStatusCode` as an error to catch is here, instead of return it as a result of running tests?

---

If a test completes without error, we should always exit `0`, if a test fails or throws and error or for whatever reason is not a clean exit, it should exit with `failStatusCode`

---

Still waiting on this.

---

I removed the function `runUntilExit`, and handled exit logic in `runCommandLine`. Since `failStatusCode` and other `ElementOptions` is passed into `runCommandLine`, I think it's better to handle exit there